pkg/process/monitor/process_monitor.go

```diff
@@ -98,6 +98,7 @@ type ProcessCallback func(pid int)
 //	                                         then the Exit callback will evaluate the same metadata on Exit.
 //	                                         We need to save the metadata here as /proc/pid doesn't exist anymore.
 func GetProcessMonitor() *ProcessMonitor {
+	processMonitor.refcount.Inc()
```

Comment:
I'm not too keen on this API because I think clients of this library can inadvertently call `GetProcessMonitor` without known it can have a side-effect, but I think it's fine to address later on.

Reply:
Do you have any other suggestion?
Calling initialize many times it problematic, so it is either here or in the initialize
regardless, we've documented that behavior

Reply:
maybe add a `clientID` argument to `GetProcessMonitor`? 🤷‍♂️